File: create_icon.js
import sharp from 'sharp';

async function createSquareIcons() {
    try {
        // Read the original logo
        const logo = sharp('public/logo.png');
        const metadata = await logo.metadata();

        console.log(`Original logo size: ${metadata.width}x${metadata.height}`);

        // Create 512x512 icon
        const size512 = 512;
        const padding512 = 60;
        const maxLogoSize512 = size512 - (2 * padding512);

        // Calculate scale to fit logo
        const scale512 = Math.min(maxLogoSize512 / metadata.width, maxLogoSize512 / metadata.height);
        const newWidth512 = Math.round(metadata.width * scale512);
        const newHeight512 = Math.round(metadata.height * scale512);

        await sharp('public/logo.png')
            .resize(newWidth512, newHeight512, { fit: 'inside' })
            .extend({
                top: Math.round((size512 - newHeight512) / 2),
                bottom: Math.round((size512 - newHeight512) / 2),
                left: Math.round((size512 - newWidth512) / 2),
                right: Math.round((size512 - newWidth512) / 2),
                background: { r: 255, g: 255, b: 255, alpha: 1 }
            })
            .png()
            .toFile('public/icon-512x512.png');

        console.log('✓ Created icon-512x512.png');

        // Create 192x192 icon
        const size192 = 192;
        const padding192 = 24;
        const maxLogoSize192 = size192 - (2 * padding192);

        const scale192 = Math.min(maxLogoSize192 / metadata.width, maxLogoSize192 / metadata.height);
        const newWidth192 = Math.round(metadata.width * scale192);
        const newHeight192 = Math.round(metadata.height * scale192);

        await sharp('public/logo.png')
            .resize(newWidth192, newHeight192, { fit: 'inside' })
            .extend({
                top: Math.round((size192 - newHeight192) / 2),
                bottom: Math.round((size192 - newHeight192) / 2),
                left: Math.round((size192 - newWidth192) / 2),
                right: Math.round((size192 - newWidth192) / 2),
                background: { r: 255, g: 255, b: 255, alpha: 1 }
            })
            .png()
            .toFile('public/icon-192x192.png');

        console.log('✓ Created icon-192x192.png');
        console.log('\nSquare icons created successfully!');

    } catch (error) {
        console.error('Error creating icons:', error);
        process.exit(1);
    }
}

createSquareIcons();

File: create_icon.py
from PIL import Image

# Open the original logo
logo = Image.open('public/logo.png')

# Create a new square image with white background
size = 512
square_img = Image.new('RGBA', (size, size), (255, 255, 255, 255))

# Calculate scaling to fit logo in square with padding
padding = 60  # pixels of padding on each side
max_logo_size = size - (2 * padding)

# Get logo dimensions
logo_width, logo_height = logo.size

# Calculate scale factor to fit logo within max size
scale = min(max_logo_size / logo_width, max_logo_size / logo_height)

# Resize logo
new_width = int(logo_width * scale)
new_height = int(logo_height * scale)
logo_resized = logo.resize((new_width, new_height), Image.Resampling.LANCZOS)

# Calculate position to center the logo
x = (size - new_width) // 2
y = (size - new_height) // 2

# Paste logo onto square background
square_img.paste(logo_resized, (x, y), logo_resized if logo_resized.mode == 'RGBA' else None)

# Save as icon-512x512.png
square_img.save('public/icon-512x512.png', 'PNG', optimize=True)

# Also create 192x192 version
square_img_192 = square_img.resize((192, 192), Image.Resampling.LANCZOS)
square_img_192.save('public/icon-192x192.png', 'PNG', optimize=True)

print("Square icons created successfully!")
print("- icon-512x512.png (512x512)")
print("- icon-192x192.png (192x192)")
